internal/proto/local/plugin/host_plugin_service.proto

```diff
@@ -0,0 +1,174 @@
+syntax = "proto3";
+
+package plugin;
+
+option go_package = "github.com/hashicorp/boundary/sdk/pbs/plugin;plugin";
+
+import "google/protobuf/struct.proto";
+
+import "controller/api/resources/hostcatalogs/v1/host_catalog.proto";
+import "controller/api/resources/hostsets/v1/host_set.proto";
+import "controller/api/resources/hosts/v1/host.proto";
+
+// HostPluginService describes the service for host plugins.
+service HostPluginService {
+  // OnCreateCatalog is a hook that runs when a host catalog is
```

Comment:
Maybe I'm off base with this observation, but happy to talk about it: Are these really "hooks"?  To me hooks suggest a runtime component relying on callbacks.  I guess you could argue that is what happens w/ plugins.  I don't know. Thoughts?

Reply:
Thinking through the process:

* It's a hook because of the idioms ("on create catalog"). The plugin has the option of sending back data to persist, and returning errors (possibly due to additional validation, etc). It also is not expected to do anything additional to facilitate creation - in fact, we likely do not expect anything to be created here (other than maybe credential rotation).
* It's not a hook because we would likely always be relying on it for initial authentication data persistence.

I think it still is more of a hook than not, mainly based on idioms and lack of pure necessity of action on part of the plugin here during creation - this could no-op and the host catalog would still be created.

Reply:
Resolving this now as Todd and I discussed this offline.